### PR TITLE
get rid of inclusion of pf17 inside of `unbounded_variant`

### PR DESCRIPTION
**Pull request by serges147**

### CODE REVIEW

File: include/cetl/unbounded_variant.hpp [hunk 1]
```diff
@@ -961,6 +961,57 @@ struct base_move<Pmr, Footprint, Copyable, true /*Movable*/, Alignment>
...
+
+/// Implementation similar to \ref std::in_place_type_t or \ref cetl::pf17::in_place_type_t.
+///
+/// In use by several \ref cetl::unbounded_variant constructors.
+///
+/// Can't use directly either of already existing `std::in_place_type_t` or `cetl::pf17::in_place_type_t` types due to
+/// C++14 limitation and polyfill optionality (by design in CETL, according to Scott), so a bit of code duplication.
+///
+template <typename T>
+struct in_place_type_t
```
Comment (thirtytwobits) [inline include/cetl/unbounded_variant.hpp:979]: Can this be a public member of unbounded_variant? `ub_var` feels like an ugly hack.

Comment (serges147) [reply include/cetl/unbounded_variant.hpp]: ok

**Review by pavel-kirienko — COMMENTED** [review]
If we're going to keep this, should we not alert the user in the doc comments for these constructors that one should consider using the freestanding factory function instead:

https://github.com/OpenCyphal/CETL/blob/4c754a5e58dcb88e06aef89ddb08c01876e07a18/include/cetl/unbounded_variant.hpp#L1109

https://github.com/OpenCyphal/CETL/blob/4c754a5e58dcb88e06aef89ddb08c01876e07a18/include/cetl/unbounded_variant.hpp#L1130

https://github.com/OpenCyphal/CETL/blob/4c754a5e58dcb88e06aef89ddb08c01876e07a18/include/cetl/unbounded_variant.hpp#L1156

https://github.com/OpenCyphal/CETL/blob/4c754a5e58dcb88e06aef89ddb08c01876e07a18/include/cetl/unbounded_variant.hpp#L1180

So that one does not have to touch the ad-hoc `in_place_type`

**Comment by serges147** [comment]
> So that one does not have to touch the ad-hoc `in_place_type`

Good idea, I will do this.

**Review by pavel-kirienko — APPROVED** [review]

**Review by thirtytwobits — APPROVED** [review]

**Pull request merged**